### DISCUSSION TITLE
How to get linetypes from layers?

### DISCUSSION THREAD

**Discussion by yingyue4008** [LibreDWG/libredwg#573] [Q&A] [answered]
Excuse me, I currently get dashes from the LTYPE of the layer, but there is linetype information in the CAD layer. Is there any deviation in this way?

**Comment by rurban** [comment, marked as answer]
if you look at the dwg.spec, you'll see that the LAYER.ltype handle contains the link to the LINETYPE.
Plus there's a LAYER.linewt (lineweight)

**Reply by yingyue4008** [reply]
"handle contains the link to the LINETYPE", I didn't get here. How did I connect,I wish I had a simple example. Thank you very much

**Comment by rurban** [comment]
A simple example would e.g. be VALUE_HANDLE for decode, because it resolves the name of a table handle when logging.

src/dec_macros.h:269
``` 
              if (dwg_ref_object_silent (dwg, ref))
                  const char *u8 = dwg_ref_tblname (dwg, ref, &alloced);
and dwg_ref_objname (dwg, ref)
```

**Reply by yingyue4008** [reply]
Thank you very much. Let me follow the above content first

**Reply by yingyue4008** [reply]
Excuse me, What should I do if I can't get the name of the print style in the same way?

**Reply by yingyue4008** [reply]
Found that ColorAlpha got a value of 0,Is this a bug?

**Comment by yingyue4008** [comment]
At present, the transparency, plot style and whether to print of the layer have not obtained valid data. Is my operation wrong?

**Reply by rurban** [reply]
Show your code

**Reply by yingyue4008** [reply]
I am the qt program. The example is as follows:

1.QString strDxf = QString::number(pLayer->color.alpha);
2.Dwg_Object* pPoltObje = getRefObjectSilent(dwg, pLayer->plotstyle);
QString strDxf = QString(QLatin1String(pPoltObje->name));
3.Print field not found

**Reply by rurban** [reply]
You need to check if  getRefObjectSilent(dwg, pLayer->plotstyle) returns non-NULL.
Not all layers do have a plotstyle defined

**Reply by yingyue4008** [reply]
I have checked that the return value of getRefObjectSilent is not empty, and this layer is styled in cad, so I think it is not obtained in the program
